### ISSUE TITLE
[Accessibility] VoiceOver focus gets trapped in semaphore buttons after hover

### ISSUE THREAD

**Issue by pachlava** [Automattic/simplenote-electron#2803]
_Only for Electron on macOS_

### Expected
The interaction with app's semaphore should be similar to some native apps experience (e.g. take macOS `Settings` for example).

The scenario described below does not occur with native apps and with some Electron apps I've tried (Spotify and GH Desktop). As soon as one of the semaphore buttons looses focus, VoiceOver should 'forget' about it.

### Observed
This is reproducible with the following setting selected in VoiceOver Utility:
![Screenshot 2021-03-23 at 20 14 20](https://user-images.githubusercontent.com/73365754/112197247-88f62780-8c14-11eb-8abc-a1a1f76b75a4.png)

See the video. As soon as the pointer hovers above one of the semaphore buttons (take `close` button for example), the focus gets trapped there, and even after clicking other controls, it will remain there (e.g. second 22, note the VoiceOver memo):

https://user-images.githubusercontent.com/73365754/112197888-39fcc200-8c15-11eb-8145-18c903d9b413.mov

Note that the VoiceOver grey box shows this even after opening `Shortcuts` dialog:
<img width="998" alt="Screenshot 2021-03-23 at 20 17 36" src="https://user-images.githubusercontent.com/73365754/112198078-6b758d80-8c15-11eb-8db0-66d75a5a317e.png">

### Reproduced
1. Open `VoiceOver Utility` ~> `Navigation` ~> enable `Synchronize keyboard focus and VoiceOver cursor` and select `Mouse pointer: Moves VoiceOver cursor`
2. Enable VoiceOver in Accessibility (macOS `Settings`)
3. Open Simplenote
4. Hover above the semaphore `close` button and note it's focused (with black rectangle), also pay attention to description given by VoiceOver:

<img width="998" alt="Screenshot 2021-03-23 at 20 17 36" src="https://user-images.githubusercontent.com/73365754/112198078-6b758d80-8c15-11eb-8db0-66d75a5a317e.png">

5. Click anywhere else in Simplenote (e.g. open `Menu` ~> `Settings`), and note that black rectangle is no longer shown for `close` semaphore button

6. Wait for VoiceOver to finish describing the Settings dialog, and then start describing `close` button again. Pressing the `CapsLock + Space` will close the app (I have CapsLock selected as VoiceOver modifier in Voice Over Utility General)

### Where did you see the bug
- System Model: MBP 16"
- OS: macOS 
- OS version: 11.2.3 (Big Sur)
- Simplenote app version: 2.9.0-beta1 Desktop
